JSQDataSourcesKit/JSQDataSourcesKit/CollectionViewDataSourceProvider.swift

```diff
@@ -32,12 +32,13 @@ A `CollectionViewDataSourceProvider` is responsible for providing a data source
 - Adding, removing, or reloading sections as the provider's `sections` are modified
 */
 public final class CollectionViewDataSourceProvider <
-    Item,
     SectionInfo: CollectionViewSectionInfo,
     CellFactory: CollectionViewCellFactoryType,
     SupplementaryViewFactory: CollectionSupplementaryViewFactoryType
-    where SectionInfo.Item == Item, CellFactory.Item == Item, SupplementaryViewFactory.Item == Item>: CustomStringConvertible {
+    where SectionInfo.Item == CellFactory.Item, CellFactory.Item == SupplementaryViewFactory.Item>: CustomStringConvertible {
 
+    public typealias Item = CellFactory.Item
```

Comment:
Minor change, let's treat `SectionInfo` as "the source of truth". I view the `Section` as *the* data, not the `CellFactory`.

````swift
where SectionInfo.Item == CellFactory.Item, SectionInfo.Item == SupplementaryViewFactory.Item
````

Same change for the `typealias` + add doc comments. :smile: 
````swift
/// The type of elements for the data source provider.
public typealias Item = SectionInfo.Item
````